### PR TITLE
test cases for service creation added

### PR DESCRIPTION
**Pull request by Shrutikabansal**
Test cases for service creation is added where checking whether service is created or not.

### CODE REVIEW

File: netcontrold/tests/lib/test_util.py [hunk 1]
```diff
@@ -262,3 +262,35 @@ def test_rr_cpu_in_numa_positive(self, mock_open):
...
+
+
+class TestUtil_test_thread_creation(TestCase):
+
+    def setUp(self):
+        util.Memoize.forgot = True
+        thread = util.Thread()
+
+    def check_thread_exist(thread):
+        if threading.activeCount() > 0:
```
Comment (gmuthukr) [inline netcontrold/tests/lib/test_util.py:274]: Can this be evaluated == 1 instead of > 1 ?, as you created only one thread.

**Comment by Shrutikabansal** [comment]
added a new test case for checking the existence of thread and as well as changes in util.py file to check whether thread exists or not using .activecount()

File: netcontrold/lib/util.py [hunk 2]
```diff
@@ -146,6 +146,10 @@ def __init__(self, shuteventobj):
         self.ncd_shutdown = shuteventobj
 
 
+def Thread_status():
```
Comment (gmuthukr) [inline netcontrold/lib/util.py:149]: Read below comment ..

File: netcontrold/tests/lib/test_util.py [hunk 3]
```diff
@@ -262,3 +262,35 @@ def test_rr_cpu_in_numa_positive(self, mock_open):
         out = util.rr_cpu_in_numa()
         expected = [0, 2, 1, 3]
         self.assertEqual(out, expected)
+
+
+class TestUtil_test_thread_creation(TestCase):
+
+    def setUp(self):
+        util.Memoize.forgot = True
+        thread = util.Thread()
```
Comment (gmuthukr) [inline netcontrold/tests/lib/test_util.py:271]: You need to inherit util.Thread like how CtlDThread does in main app. For eg:

```
class DummyNcdThread(util.Thread):
    def __init__(self, eobj):
        util.Thread.__init__(self, eobj)

    def run(self):
        <set a dummy variable to 1>
       while (not self.ncd_shutdown.is_set()):
          <do some no_op with some sleep interval not to hog cpu>
       return

class TestUtil_test_thread_creation(TestCase):

  def setUp(self):
        util.Memoize.forgot = True

  def check_thread_exist(self):
        stop_event = threading.Event()
        tobj = DummyNcdThread(stop_event)
        tobj.start()
        tobj.stop_event.set()
        tobj.join()
       <assert dummy variable set by thread equal to 1>

  def check_thread_no_exist(self):
     < think of thread failure case >

```

File: netcontrold/tests/lib/test_util.py [hunk 4]
```diff
@@ -262,3 +262,35 @@ def test_rr_cpu_in_numa_positive(self, mock_open):
...
+        else:
+            value1 = 'F'
+
+        count = util.Thread_status()
+        if count > 0:
+            value2 = 'T'
+        else:
+            value2 = 'F'
+
+        self.assertEqual(value1, value2)
```
Comment (gmuthukr) [inline netcontrold/tests/lib/test_util.py:285]: Although you will rewrite test, just fyi the flaw in the above assert: when both are 'F', the test will still pass.

File: netcontrold/tests/lib/test_util.py [hunk 5]
```diff
@@ -262,3 +262,35 @@ def test_rr_cpu_in_numa_positive(self, mock_open):
...
+
+
+class TestUtil_test_searvice_creation(TestCase):
+
+    def setUp(self):
+        util.Memoize.forgot = True
+        service = util.Service()
```
Comment (gmuthukr) [inline netcontrold/tests/lib/test_util.py:292]: Creating service without call back did not result "("Function callback can not be empty to start new service.")" error ?

File: netcontrold/tests/lib/test_util.py [hunk 6]
```diff
@@ -262,3 +262,35 @@ def test_rr_cpu_in_numa_positive(self, mock_open):
...
+
+    def check_pid_exist():
```
Comment (gmuthukr) [inline netcontrold/tests/lib/test_util.py:294]: Like in above thread tests I showed, create few service tests by actually using 'dummy call back' as we did in ncd_ctl and test its existence and expected dummy op etc.

**Review by gmuthukr — CHANGES_REQUESTED** [review]

File: netcontrold/tests/lib/test_util.py (hunk 5 above)
Comment (Shrutikabansal) [reply netcontrold/tests/lib/test_util.py:292]: No, No error popped up.

**Pull request closed without merging**